dockers/docker-database/docker-database-init.sh

```diff
@@ -1,5 +1,26 @@
 #!/usr/bin/env bash
 
+# For linux host namespace, in both single and multi ASIC platform use the loopback interface
+# For other namespaces, use eth0 interface which is connected to the docker0 bridge in the host.
+if [[ $NAMESPACE_ID == "" ]]
+then
+    INTFC=lo
+else
+    INTFC=eth0
+fi
+
+# Get the ip address of the interface
+# if the ip address was not retrieved correctly, put localhost(127.0.0.1) as the default.
+host_ip=$(ip -4 -o addr show $INTFC | awk '{print $4}' | cut -d'/' -f1 | head -1)
+if [[ $host_ip == "" ]]
+then
+    host_ip=127.0.0.1
+fi
+
+# Export the host_ip as environment variable used in database_config.json.j2 to derive the hostname
+# as defined "hostname" : "{{HOST_IP}}"
+export HOST_IP=$host_ip
```

Comment:
Instead export it globally, you could just add env var to one command line.
```
    HOST_IP=$host_ip j2 /usr/share/sonic/templates/database_config.json.j2 > $REDIS_DIR/sonic-db/database
```

Reply:
I have updated, with HOST_IP on command line itself.